x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/AuthenticationService.java

```diff
@@ -222,6 +219,13 @@ public void authenticate(
         authenticatorChain.authenticateAsync(context, listener);
     }
 
+    /**
+     * This must be called after the HTTP request has been authenticated.
```

Comment:
nit: not sure if there is easy access ... but if there is... can you add an assert in the code that authentication bits are present ?

Reply:
It's easier to assert authentication in the called `AuditTrail#authenticationSuccess`, if I get you right.